plugins/Live/Live.php

```diff
@@ -57,69 +52,99 @@ public function addJsGlobalVariables(&$out)
         ";
     }
 
-    public static function isVisitorLogEnabled($idSite = null)
-    {
-        [$profileEnabled, $logEnabled] = self::getSettings($idSite);
-
-        return $logEnabled;
-    }
-
-    public static function isVisitorProfileEnabled($idSite = null)
+    /**
+     * Throws an exception if visits log is disabled
+     *
+     * @param null|int|array $idSite
+     * @throws \Exception
+     */
+    public static function checkIsVisitorLogEnabled($idSite = null): void
     {
-        [$profileEnabled, $logEnabled] = self::getSettings($idSite);
+        $systemSettings = new SystemSettings();
 
-        return $profileEnabled;
-    }
+        if ($systemSettings->activateVisitorLog->getValue() == false) {
+            throw new \Exception('Visits log is deactivated globally. A user with super user access can enable this feature in the general settings.');
+        }
 
-    private static function getSettings($idSite = null)
-    {
         if (empty($idSite)) {
             $idSite = Common::getRequestVar('idSite', 0, 'int');
         }
 
-        $cache = Cache::getTransientCache();
-        $siteIdLoaded = $cache->fetch(self::CurrentSiteCacheKey);
-        $visitorProfileCached = $cache->contains(self::ProfileEnabledCacheKey);
-        $visitorLogCached = $cache->contains(self::LogEnabledCacheKey);
+        if (!empty($idSite)) {
+            $idSites = is_array($idSite) ? $idSite : [$idSite];
 
-        if ($visitorProfileCached && $visitorLogCached && $idSite == $siteIdLoaded) {
-            return [
-                $cache->fetch(self::ProfileEnabledCacheKey),
-                $cache->fetch(self::LogEnabledCacheKey),
-            ];
-        }
+            foreach ($idSites as $idSite) {
+                $settings = new MeasurableSettings($idSite);
 
-        $siteIdLoaded = $idSite;
-        $visitorProfileEnabled = true;
-        $visitorLogEnabled = true;
+                if ($settings->activateVisitorLog->getValue() == false) {
+                    throw new \Exception('Visits log is deactivated in website settings. A user with at least admin access can enable this feature in the settings for this website.');
```

Comment:
could we include the idSite in here? This may be useful when multiple sites are given. Ideally, if multiple sites are given, we would only filter out the idSites that aren't working though and keep all others but for a first release I would say this isn't needed. Be just useful to know which idSite of the many are disabled

Reply:
The filtering can't be done in this method. But can add it to the `Live.getLastVisitsDetails`